evap/staff/views.py

```diff
@@ -538,6 +545,37 @@ def course_email(request, semester_id, course_id):
 
 
 @staff_required
+def course_import_participants(request, semester_id, course_id):
+    course = get_object_or_404(Course, id=course_id)
+    raise_permission_denied_if_archived(course)
+
+    form = UserImportForm(request.POST or None, request.FILES or None)
+
+    if form.is_valid():
+        operation = request.POST.get('operation')
+        if operation not in ('test', 'import'):
+            raise SuspiciousOperation("Invalid POST operation")
+
+        # Extract data from form.
+        excel_file = form.cleaned_data['excel_file']
+
+        test_run = operation == 'test'
+
+        # Parse table.
+        imported_users = UserImporter.process(request, excel_file, test_run)
+
+        # Test run, or an error occurred while parsing -> stay and display error.
+        if test_run or not imported_users:
+            return render(request, "staff_import_participants.html", dict(course=course, form=form))
```

Comment:
unfortunately this is very implicit. UserImporter.process() sets message.errors which will be displayed on the same website in the case that imported_users == None instead of the semester_view page.